anthos-bm-gcp-terraform/main.tf

```diff
@@ -0,0 +1,186 @@
+    { for vm in module.controlplane_vm_hosts.vm_info : vm.hostname => vm.externalIp },
+    { for vm in module.worker_vm_hosts.vm_info : vm.hostname => vm.externalIp }
+  )
+}
+
+module "enable_google_apis_primary" {
+  source                      = "terraform-google-modules/project-factory/google//modules/project_services"
+  version                     = "10.2.0"
+  project_id                  = var.project_id
+  activate_apis               = var.primary_apis
+  disable_services_on_destroy = false
+}
+
+module "enable_google_apis_secondary" {
+  source  = "terraform-google-modules/project-factory/google//modules/project_services"
+  version = "10.2.0"
+  # fetched from previous module to explicitely express dependency
+  project_id                  = module.enable_google_apis_primary.project_id
+  depends_on                  = [module.enable_google_apis_primary]
+  activate_apis               = var.secondary_apis
+  disable_services_on_destroy = false
+}
+
+module "create_service_accounts" {
+  source  = "terraform-google-modules/service-accounts/google"
+  version = "~> 3.0"
+  # fetched from previous module to explicitely express dependency
+  project_id = module.enable_google_apis_secondary.project_id
+  depends_on = [
+    module.enable_google_apis_primary,
+    module.enable_google_apis_secondary
+  ]
+  names         = ["baremetal-gcr"]
+  generate_keys = true
+  project_roles = [
+    "${var.project_id}=>roles/gkehub.connect",
+    "${var.project_id}=>roles/gkehub.admin",
+    "${var.project_id}=>roles/logging.logWriter",
+    "${var.project_id}=>roles/monitoring.metricWriter",
+    "${var.project_id}=>roles/monitoring.dashboardEditor",
+    "${var.project_id}=>roles/stackdriver.resourceMetadata.writer",
+  ]
+}
+
+module "instance_template" {
+  source = "terraform-google-modules/vm/google//modules/instance_template"
+  depends_on = [
+    module.enable_google_apis_primary,
+    module.enable_google_apis_secondary
+  ]
+  # fetched from previous module to explicitely express dependency
+  project_id           = module.enable_google_apis_secondary.project_id
+  region               = var.region          # --zone=${ZONE}
+  service_account      = var.service_account # --scopes cloud-platform
+  source_image_family  = var.image_family    # --image-family=ubuntu-2004-lts
+  source_image_project = var.image_project   # --image-project=ubuntu-os-cloud
+  machine_type         = var.machine_type    # --machine-type $MACHINE_TYPE
+  disk_size_gb         = var.boot_disk_size  # --boot-disk-size 200G
+  disk_type            = var.boot_disk_type  # --boot-disk-type pd-ssd
+  can_ip_forward       = true                # --can-ip-forward
+  network              = var.network         # --network default
+  tags                 = var.tags            # --tags http-server,https-server
+  # TODO:: Unavailable as of now
+  # min_cpu_platform = var.min_cpu_platform # --min-cpu-platform "Intel Haswell"
```

Comment:
- Raise why `min_cpu_platform` is left out in [this module](https://github.com/terraform-google-modules/terraform-google-vm/tree/master/modules/instance_template). If allowed open a PR to support it.

- Open a PR to select GPU options on the `instance_template` module

Reply:
[PR#160](https://github.com/terraform-google-modules/terraform-google-vm/pull/160) and [PR#161](https://github.com/terraform-google-modules/terraform-google-vm/pull/161)